### PR TITLE
Sometimes, people have secure headers but opt out of CSP

### PR DESCRIPTION
**Pull request by jcmuller**

### CODE REVIEW

File: lib/rollbar/middleware/js.rb [hunk 1]
```diff
@@ -133,6 +133,7 @@ def html_safe_if_needed(string)
       def append_nonce?
         defined?(::SecureHeaders) && ::SecureHeaders.respond_to?(:content_security_policy_script_nonce) &&
           defined?(::SecureHeaders::Configuration) &&
+          !::SecureHeaders::Configuration.get.csp.opt_out? &&
```
Comment (jondeandres) [inline lib/rollbar/middleware/js.rb]: This works for secure headers 2.x and 3.x?

Comment (jcmuller) [inline lib/rollbar/middleware/js.rb]: Only tested it against 3.x

Comment (jondeandres) [inline lib/rollbar/middleware/js.rb]: Nice, thanks for the details. We'll use this PR to fix the problem but we'll check first the behavior on  2.x.

Comment (jcmuller) [inline lib/rollbar/middleware/js.rb]: The similar check on 2.x would be `SecureHeaders::Configuration.csp.nil?`. Version detection is necessary, alas, from what I've gathered.

**Comment by coryvirok** [comment]
@jondeandres can we merge and release this?

**Comment by jondeandres** [comment]
I will sync this branch with master, check the tests pass and merge

**Comment by jcmuller** [comment]
This branch is up to date with current master (7845d23). (Last commit from 4 months ago).

**Comment by jcmuller** [comment]
Sorry, I was wrong before. Updated to latest upstream master and rebased off of it.

**Comment by jcmuller** [comment]
Any news on this?

**Comment by jcmuller** [comment]
Hi guys, It's been over 6 months. Is this going to be merged?

**Comment by rokob** [comment]
I'll see what I can do to make the necessary changes that will allow this to work for both 2.x and 3.x and get this merged this week.

File: lib/rollbar/middleware/js.rb (hunk 1 above)
Comment (rokob) [inline lib/rollbar/middleware/js.rb]: To be fair, we already do not support 2.x because of the next line we are already using `Configuration.get`

**Comment by rokob** [comment]
I took this over in #629 and merged that, so closing this. Sorry but that was the simplest way to move this forward.

**Pull request closed without merging**